src/VisualStudio/Core/Def/Implementation/TableDataSource/VisualStudioBaseDiagnosticListTable.LiveTableDataSource.cs

```diff
@@ -175,6 +172,19 @@ public override AbstractTableEntriesSource<DiagnosticData> CreateTableEntriesSou
                 return new TableEntriesSource(this, item.Workspace, item.ProjectId, item.DocumentId, item.Id);
             }
 
+            private void ConnectToDiagnosticService(Workspace workspace, IDiagnosticService diagnosticService)
+            {
+                if (diagnosticService == null)
+                {
+                    // it can be null in unit test
+                    return;
+                }
+
+                _diagnosticService.DiagnosticsUpdated += OnDiagnosticsUpdated;
+
+                PopulateInitialData(workspace, diagnosticService);
+            }
+
```

Comment:
This whole file can probably reverted right?

Reply:
no this is valid change.